accelerator/apis.c

```diff
@@ -153,55 +153,153 @@ status_t api_find_transactions_obj_by_tag(
   return ret;
 }
 
+static int _idx_sort(void const* lhs, void const* rhs) {
+  iota_transaction_t* _lhs = (iota_transaction_t*)lhs;
+  iota_transaction_t* _rhs = (iota_transaction_t*)rhs;
+
+  return (transaction_current_index(_lhs) < transaction_current_index(_rhs))
+             ? -1
+             : (transaction_current_index(_lhs) >
+                transaction_current_index(_rhs));
+}
+
+static void _get_first_bundle_from_transactions(
+    transaction_array_t* const transactions,
+    bundle_transactions_t* const bundle) {
+  iota_transaction_t* tail = NULL;
+  iota_transaction_t* curr_tx = NULL;
+  iota_transaction_t* prev = NULL;
+
+  utarray_sort(transactions, _idx_sort);
+  tail = (iota_transaction_t*)utarray_eltptr(transactions, 0);
+  bundle_transactions_add(bundle, tail);
+
+  prev = tail;
+  TX_OBJS_FOREACH(transactions, curr_tx) {
+    if (transaction_current_index(curr_tx) ==
+            (transaction_current_index(prev) + 1) &&
+        (memcmp(transaction_hash(curr_tx), transaction_trunk(prev),
+                FLEX_TRIT_SIZE_243) == 0)) {
+      bundle_transactions_add(bundle, curr_tx);
+      prev = curr_tx;
+    }
+  }
+}
+
+static void _recv_init_client_service(iota_client_service_t* const serv,
+                                      char const* const host,
+                                      uint16_t const port) {
+  serv->http.path = "/";
+  serv->http.content_type = "application/json";
+  serv->http.accept = "application/json";
+  serv->http.host = host;
+  serv->http.port = port;
+  serv->http.api_version = 1;
+  serv->http.ca_pem = NULL;
+  serv->serializer_type = SR_JSON;
+  iota_client_core_init(serv);
+}
+
+static transaction_array_t* _get_bundle_transactions(
+    iota_client_service_t* const serv, find_transactions_req_t* recv_req,
+    flex_trit_t const* const bundle_hash) {
+  flex_trit_t bundle_hash_flex[FLEX_TRIT_SIZE_243];
+  transaction_array_t* out_tx_objs = transaction_array_new();
+  recv_req->approvees = NULL;
+  recv_req->bundles = NULL;
+  recv_req->tags = NULL;
+  recv_req->addresses = NULL;
+
+  flex_trits_from_trytes(bundle_hash_flex, NUM_TRITS_BUNDLE, bundle_hash,
+                         NUM_TRITS_HASH, NUM_TRYTES_BUNDLE);
```

Comment:
What is this conversion for?
`bundle_hash_flex` is not used in the following statement, plus `bundle_hash` is not trytes.